packages/runner-shared/src/event-manager.js

```diff
@@ -482,13 +484,23 @@ export const eventManager = {
       localBus.emit('script:error', err)
     })
 
-    Cypress.on('test:before:run:async', (test) => {
+    Cypress.on('test:before:run:async', (attr, test) => {
       if (studioRecorder.suiteId) {
         studioRecorder.setTestId(test.id)
       }
 
-      if (studioRecorder.hasRunnableId && test.invocationDetails) {
-        studioRecorder.setFileDetails(test.invocationDetails)
+      if (studioRecorder.hasRunnableId) {
+        if (test.invocationDetails) {
+          studioRecorder.setFileDetails(test.invocationDetails)
+        }
+
+        if (studioRecorder.suiteId) {
+          if (test.parent && test.parent.id !== 'r1') {
+            studioRecorder.setRunnableTitle(test.parent.title)
+          }
+        } else {
+          studioRecorder.setRunnableTitle(test.title)
+        }
       }
```

Comment:
I think it would be good to abstract these lines into a `studioRecorder` method or another utility. We should try to keep the responsibility of this file to managing events and move any logic into separate places.

Reply:
Could you move these lines (486-502) into `studioRecorder` as well? 

I think the litmus test for abstracting out of this file is whether there is `if/else` logic involved. There are a couple other places in this file that still have `if/else`, but they're unrelated to this PR, so of course don't need to be taken care of now.